ipatests/test_integration/test_epn.py

```diff
@@ -344,22 +353,117 @@ def test_EPN_smoketest_3(self):
         expected_users = ["user1", "user3", "user7", "user14", "user28"]
         assert sorted(user_lst) == sorted(expected_users)
 
-    def test_EPN_nbdays(self):
+    def test_EPN_nbdays_0(self, cleanupmail):
         """Test the to/from nbdays options (implies --dry-run)
 
            We have a set of users installed with varying expiration
            dates. Confirm that to/from nbdays finds them.
+
+           Make sure --dry-run does not accidentally send emails.
         """
 
-        # Compare the notify_ttls values
+        # Use the notify_ttls values with a 1-day sliding window
         for i in self.notify_ttls:
             user_list = []
-            (stdout_text_client, unused) = self._check_epn_output(
-                self.clients[0], from_nbdays=i, to_nbdays=i + 1, dry_run=True)
+            (stdout_text_client, unused, _unused) = self._check_epn_output(
+                self.clients[0], from_nbdays=i, to_nbdays=i + 1, dry_run=True
+            )
             for user in json.loads(stdout_text_client):
                 user_list.append(user["uid"])
             assert len(user_list) == 1
-            assert user_list[0] == "user%d" % i
+            userid = "user{id}".format(id=i)
+            assert user_list[0] == userid
+
+            # Check that the user list is expected for any given notify_ttls.
+            (stdout_text_client, unused, _unused) = self._check_epn_output(
+                self.clients[0], to_nbdays=i
+            )
+            user_list = [user["uid"] for user in json.loads(stdout_text_client)]
+            assert len(user_list) == 1
+            assert user_list[0] == "user{id}".format(id=i - 1)
+
+            # make sure no emails were sent
+            result = self.clients[0].run_command(['ls', '-lha', '/var/mail/'])
+            assert userid not in result.stdout_text
+
+    def test_EPN_nbdays_1(self, cleanupmail):
+        """Test that for a given range, we find the users in that range"""
+
+        # Use hardcoded date ranges for now
+        for date_range in [(0, 5), (7, 15), (1, 20)]:
+            expected_user_list = ["user{i}".format(i=i)
+                                  for i in range(date_range[0], date_range[1])]
+            (stdout_text_client, unused, _unused) = self._check_epn_output(
+                self.clients[0],
+                from_nbdays=date_range[0],
+                to_nbdays=date_range[1]
+            )
+            user_list = [user["uid"] for user in json.loads(stdout_text_client)]
+            for user in expected_user_list:
+                assert user in user_list
+            for user in user_list:
+                assert user in expected_user_list
+
+    # Test the to/from nbdays options behavior with illegal input
+
+    def test_EPN_nbdays_input_0(self):
```

Comment:
If you use cleanupmail for the other --dry-run tests shouldn't it be here for consistency? Or in none of them?

Reply:
I don't think so, because test_EPN_nbdays_0 would trip if --dry-run didn't work as expected.